lib/puppet/transaction.rb

```diff
@@ -79,8 +79,12 @@ def perform_pre_run_checks
   # necessary events.
   def evaluate(&block)
     block ||= method(:eval_resource)
-    generator = AdditionalResourceGenerator.new(@catalog, relationship_graph, @prioritizer)
-    @catalog.vertices.each { |resource| generator.generate_additional_resources(resource) }
+    generator = nil
+    additionResourceGenerationTime = thinmark do
```

Comment:
Nit picky comment: the preferred naming convention for local variables is to use `snake_case`.